src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const clearing = 'clear';
+  let newState = { ...state };
+  const arr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case add:
+        Object.assign(newState, action.extraData);
+        arr.push({ ...newState });
+        break;
+
+      case remove:
+        for (const key of action.keysToRemove) {
+          if (newState[key]) {
+            delete newState[key];
+          };
+        };
+        arr.push({ ...newState });
```

Comment:
you can push just once on each iteration

Reply:
Fix it. 
![image](https://user-images.githubusercontent.com/97547824/216815934-83fcfbda-bf3c-4132-841d-d4813ed96eef.png)
If you have any questions feel free to ask in Slack